### PR TITLE
Allow instrumented Spark trace linked to Openlineage originated context

### PR DESCRIPTION
**Pull request by yiliangzhou**
# What Does This Do
If Openlineage context (parent job namespace, job name, run id) is detected in the Spark configuration, the Spark instrumentation generates the TraceID and SpanId for the root Span in a certain way such that it matches the upstream's cue, allowing the Trace from upstream task and the Spark trace linked together in Datadog.

# Motivation

# Additional Notes

# Contributor Checklist

- [ ] Format the title [according the contribution guidelines](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#title-format)
- [ ] Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any usefull labels](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#labels)
- [ ] Squash your commits prior merging or merge using GitHub's [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
- [ ] Don't use `close`, `fix` or any [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue.  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- [ ] Update the [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) in case of new configuration flag or behavior

Jira ticket: [PROJ-IDENT]

### CODE REVIEW

**Review by kr-igor — APPROVED** [review]

File: dd-java-agent/instrumentation/spark/src/testFixtures/groovy/datadog/trace/instrumentation/spark/OpenlineageParentContextTest.groovy [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package datadog.trace.instrumentation.spark
+
+import datadog.trace.api.DDSpanId
+import org.apache.spark.SparkConf
+import spock.lang.Specification
+
+class OpenlineageParentContextTest extends Specification {
```
Comment (PerfectSlayer) [inline dd-java-agent/instrumentation/spark/src/testFixtures/groovy/datadog/trace/instrumentation/spark/OpenlineageParentContextTest.groovy:7]: Should we have test cases with invalid data? Like data that won't decode properly?

**Review by PerfectSlayer — APPROVED** [review]
Few minor comments from Language Platform POV. 
You might want a review from Integrations and Data Modelling too

File: dd-java-agent/instrumentation/spark/src/main/java/datadog/trace/instrumentation/spark/OpenlineageParentContext.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    return spanId;
+  }
+
+  public long getChildRootSpanId() {
+    return childRootSpanId;
+  }
+
+  @Override
+  public AgentTraceCollector getTraceCollector() {
+    return null;
```
[the remark this answers is not in the document]
Comment (yiliangzhou) [reply dd-java-agent/instrumentation/spark/src/main/java/datadog/trace/instrumentation/spark/OpenlineageParentContext.java]: done

File: dd-java-agent/instrumentation/spark/src/main/java/datadog/trace/instrumentation/spark/OpenlineageParentContext.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  }
+
+  @Override
+  public int getSamplingPriority() {
+    return PrioritySampling.USER_KEEP;
+  }
+
+  @Override
+  public Iterable<Map.Entry<String, String>> baggageItems() {
+    return null;
```
[the remark this answers is not in the document]
Comment (yiliangzhou) [reply dd-java-agent/instrumentation/spark/src/main/java/datadog/trace/instrumentation/spark/OpenlineageParentContext.java]: done

File: dd-java-agent/instrumentation/spark/src/testFixtures/groovy/datadog/trace/instrumentation/spark/OpenlineageParentContextTest.groovy (hunk 1 above)
Comment (yiliangzhou) [reply dd-java-agent/instrumentation/spark/src/testFixtures/groovy/datadog/trace/instrumentation/spark/OpenlineageParentContextTest.groovy:7]: added validation for incoming runId to be a valid UUID, added test cases for invalid runID and missing fields.

**Review by PerfectSlayer — APPROVED** [review]
LGTM! Thanks for taking into account my comments 👍

**Comment by amarziali** [comment]
Removing IDM since there does not seem to have any changes on our owned files

**Comment by PerfectSlayer** [comment]
Did the bot just created a bunch of label on the repo by himself? 🤔

**Pull request merged**